gnat2goto/driver/tree_walk.adb

```diff
@@ -2723,28 +2723,63 @@ package body Tree_Walk is
    --    return R;
    function Do_Bit_Op (N : Node_Id;
                        Operator : Bit_Operand_Constructor)
-                      return Irep is
-      LHS_Bool_Value : constant Irep := Do_Expression (Left_Opnd (N));
-      RHS_Bool_Value : constant Irep := Do_Expression (Right_Opnd (N));
-      Cast_LHS_To_Integer : constant Irep :=
-        Make_Op_Typecast (Op0 => LHS_Bool_Value,
-                          Source_Location => Sloc (N),
-                          I_Type => Make_Signedint_Type (32));
-      Cast_RHS_To_Integer : constant Irep :=
-        Make_Op_Typecast (Op0 => RHS_Bool_Value,
-                          Source_Location => Sloc (N),
-                          I_Type => Make_Signedint_Type (32));
-      R : constant Irep := Operator (Lhs => Cast_LHS_To_Integer,
-                                     Rhs => Cast_RHS_To_Integer,
-                                     Source_Location => Sloc (N),
-                                     Overflow_Check => False,
-                                     Range_Check => False,
-                                     I_Type =>
-                                       Get_Type (Cast_LHS_To_Integer));
+                       return Irep is
+      Source_Loc : constant Source_Ptr := Sloc (N);
+      LHS_Value : constant Irep := Do_Expression (Left_Opnd (N));
+      RHS_Value : constant Irep := Do_Expression (Right_Opnd (N));
+
+      Ret_Type : constant Irep := Do_Type_Reference (Etype (N));
+      Followed_Type : constant Irep :=
+        Follow_Symbol_Type (Ret_Type, Global_Symbol_Table);
+
    begin
-      return Make_Op_Typecast (Op0 => R,
-                               Source_Location => Sloc (N),
-                               I_Type => Make_Bool_Type);
+      if Kind (Followed_Type) = I_Ada_Mod_Type then
```

Comment:
Would it make sense to do this part in a separate function?

Reply:
Done.